src/qibolab/pulses.py

```diff
@@ -660,429 +672,80 @@ def __repr__(self):
-                self._finish = self._start + self._duration
-
-    @property
-    def finish(self) -> int:
-        """Returns the time when the pulse is scheduled to finish.
-
-        Calculated as pulse.start - pulse finish.
-        """
-        if isinstance(self._finish, se_int):
-            return self._finish.value
-        return self._finish
-
-    @property
-    def se_start(self) -> se_int:
-        """Returns a symbolic expression for the pulse start."""
-
-        if not isinstance(self._start, se_int):
-            self._start = se_int(self._start)["_p" + str(self._id) + "_start"]
-        return self._start
-
-    @property
-    def se_duration(self) -> se_int:
-        """Returns a symbolic expression for the pulse duration."""
-
-        if not isinstance(self._duration, se_int):
-            self._duration = se_int(self._duration)["_p" + str(self._id) + "_duration"]
-        return self._duration
-
-    @property
-    def se_finish(self) -> se_int:
-        """Returns a symbolic expression for the pulse finish."""
-
-        if not isinstance(self._finish, se_int):
-            self._finish = se_int(self._finish)["_p" + str(self._id) + "_finish"]
-        return self._finish
-
-    @property
-    def amplitude(self) -> float:
-        """Returns the amplitude of the pulse.
-
-        Pulse amplitudes are normalised between -1 and 1.
-        """
-
-        return self._amplitude
-
-    @amplitude.setter
-    def amplitude(self, value):
-        """Sets the amplitude of the pulse.
+    The value has to be in the range [10e6 to 300e6].
+    """
+    relative_phase: Optional[float] = None
+    """Relative phase of the pulse, in radians."""
+    shape: Optional[PulseShape] = None
+    """Pulse shape, as a PulseShape object.
 
-        Args:
-            value (int | float | np.floating): a unitless value between -1 and 1.
-        """
+    See
+    :py: mod:`qibolab.pulses` for list of available shapes.
+    """
+    channel: int = 0
+    """Channel on which the pulse should be played.
 
-        if isinstance(value, int):
-            value = float(value)
-        if not isinstance(value, (float, np.floating)):
-            raise TypeError(
-                f"amplitude argument type should be float, got {type(value).__name__}"
-            )
-        if not ((value >= -1) & (value <= 1)):
-            raise ValueError(f"amplitude argument must be >= -1 & <= 1, got {value}")
-        if isinstance(value, np.floating):
-            self._amplitude = float(value)
-        elif isinstance(value, float):
-            self._amplitude = value
+    When a sequence of pulses is sent to the platform for execution,
+    each pulse is sent to the instrument responsible for playing pulses
+    the pulse channel. The connection of instruments with channels is
+    defined in the platform runcard.
+    """
+    type: PulseType = PulseType.DRIVE
+    """Pulse type, as an element of PulseType enumeration."""
+    qubit: Optional[int] = 0
```

Comment:
That's a good point, thanks. In fact, as we discussed before, I am not sure if `Pulse` need to have both a `channel` and a `qubit` attribute. I believe the `channel` should be sufficient but this would require a refactoring in most of the drivers. So I wouldn't spend much time trying to fix this cycle as `qubit` may not be needed at all.

As for native.py, at some point I would consider dropping it altogether as it is duplicating pulses for the most part. It serves as an intermediate between the runcard and pulses, but the only thing that does not allow us to instantiate `Pulse` objects directly from the runcard is the `start` time, which we can also drop once we start using delays.

Anyway, for now I agree with keeping things as it is. We could just still do since we use `Optional`
```suggestion
    qubit: Optional[int] = None
```
but I am not sure if that breaks something.

Reply:
> but I am not sure if that breaks something.

This I'm not sure as well. I'm just lazy about something that we might drop soon, and I already tested in this configuration (actually, if the labels were just working ootb it would not take too much to test the change... but for this PR they are not...).
I will just drop `Optional[]`, since with the default to `0` we won't use `None` at all.

Reply:
Last part implemented in 6357d01f0183fef3a7d5912cca743f5bb2b187f0